docs/content/4.sdk/3.advanced/extending-module.md

```diff
@@ -260,6 +262,50 @@ import { sdk } from './sdk';
 const product = await sdk.sapcc.getProductBySku('product-sku');
 ```
 
+In the example below, we are using the `extend` attribute to create a new method `getProductBySkuWithLogger` that logs the used sku and then calls the original `getProductBySku` method.
+
+`buildModule` function provides the parent object that contains the original methods and optional context. You can use it to call the original method or access additional properties in the context.
+
+We pass a factory function as the third argument to the `buildModule` function. The factory function receives the options and the parent object as arguments. Parent object contains the original methods and optional context.
+
+In this example, we are also using the `customMethod` that calls an external API using the `context.client` method.
+
+```ts
+const sdkConfig = {
+  sapcc: buildModule(
+    // sapccModule is the module that we want to extend
+    sapccModule, 
+    // module options
+    { url: "some-url" }, 
+    // factory function that receives the options and the parent object as arguments
+    (options, parent) => { 
```

Comment:
wdyt about adding an example how to define the factory function outside the `buildModule`? When customer will have a lot of modules, he may want to move the factory to a variable and then pass this variable to the buildModule